apps/arweave/src/ar_mining_worker.erl

```diff
@@ -333,24 +370,24 @@ process_sub_chunk(chunk1, Candidate, SubChunk, State) ->
 process_sub_chunk(chunk2, Candidate, SubChunk, State) ->
 	#mining_candidate{ session_key = SessionKey } = Candidate,
 	Candidate2 = Candidate#mining_candidate{ chunk2 = SubChunk },
-	case cycle_sub_chunk_cache(Candidate2, {chunk2, SubChunk}, State) of
-		{{chunk1, Chunk1, H1}, State2} ->
+	case cycle_sub_chunk_cache(Candidate2, SubChunk, #{chunk2 => true}, State) of
+		{{<<>>, #{chunk1 := true, h1 := H1}}, State2} ->
+			ar_mining_hash:compute_h2(self(), Candidate2#mining_candidate{ h1 = H1 }),
+			%% Decrement 1 for chunk2:
```

Comment:
The `Decrement xxx` comments may no longer be helpful, they are from when we managed the chunk cache size manually. I believe it's not handled transparently by ar_chunk_cache via the `chunk_cache_size_bytes` and `reserved_chunk_cache_size_bytes` fields